src/controller.py

```diff
@@ -0,0 +1,75 @@
+import os
+import subprocess
+import argparse
+import sys
+import re
+
+
+def extract_args():
+    """ Extract and return the required arguments using argparse. """
+    parser = argparse.ArgumentParser(description='Generate Living Documentation from GitHub repositories.')
+
+    parser.add_argument('--github-token', required=True, help='GitHub token for authentication.')
+    parser.add_argument('--project-state-mining', required=True, help='Enable or disable mining of project state data.')
+    parser.add_argument('--projects-title-filter', required=True, help='Filter projects by titles. Provide a list of project titles.')
+    parser.add_argument('--milestones-as-chapters', required=True, help='Treat milestones as chapters in the generated documentation.')
+    parser.add_argument('--repositories', required=True, help='JSON string defining the repositories to be included in the documentation generation.')
+
+    args = parser.parse_args()
+
+    env_vars = {
+        'GITHUB_TOKEN': args.github_token,
+        'PROJECT_STATE_MINING': args.project_state_mining,
+        'PROJECTS_TITLE_FILTER': args.projects_title_filter,
+        'MILESTONES_AS_CHAPTERS': args.milestones_as_chapters,
+        'REPOSITORIES': args.repositories
+    }
+
+    return env_vars
+
+
+def run_script(script_name, env_vars):
+    """ Helper function to run a Python script with environment variables using subprocess """
+    try:
+        # Setting up the environment for subprocess
+        env = os.environ.copy()
+        env.update(env_vars)
+
+        # Running the script with updated environment
+        result = subprocess.run(['python3', script_name], env=env, text=True, capture_output=True, check=True)
+        print(f"Output from {script_name}: {result.stdout}")
+
+    except subprocess.CalledProcessError as e:
+        print(f"Error running {script_name}: \n{e.stdout}\n{e.stderr}")
+        sys.exit(1)
+
+
+def main():
+    print("Extracting arguments from command line.")
+    env_vars = extract_args()
+
+    # Clean environment before mining
+    print("Data mining for Living Documentation")
+    run_script('clean_env_before_mining.py', env_vars)
```

Comment:
hardcoding python script name like this is definitely weird. I'd just import and call it directly as a standard Python call

Reply:
I have prepared a solution proposal for the clean script - see commit 56c6fc915d1a0f570f50306754a81381df6db610.
@MobiTikula, Can you address the refactoring of the rest of the scripts in SPIKE?